projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php

```diff
@@ -0,0 +1,121 @@
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Status\Visitor;
+
+/**
+ * Class Jetpack_Google_Drive_Helper
+ */
+class Jetpack_Google_Drive_Helper {
+	/**
+	 * Checks if the user has a valid connection to Google Drive
+	 *
+	 * @param int $user_id The user ID.
+	 * @return array       Array with single 'valid' (bool) entry.
+	 */
+	public static function validate_connection( $user_id ) {
+		$site_id = self::get_site_id();
+		if ( is_wp_error( $site_id ) ) {
+			return false;
+		}
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			// check for gdrive helper class, call synchronously on .com
+			require_lib( 'google-sheets-helper' );
+			return wp_json_encode( WPCOM_Google_Sheets_Helper::get_validate_connection_response( (int) $user_id ) );
+		}
+
+		$request_path  = sprintf( '/sites/%d/google-drive/connection', $site_id );
+		$wpcom_request = Client::wpcom_json_api_request_as_user(
+			$request_path,
+			'2',
+			array(
+				'method'  => 'GET',
+				'headers' => array(
+					'content-type'    => 'application/json',
+					'X-Forwarded-For' => ( new Visitor() )->get_ip( true ),
+				),
+			)
+		);
+
+		$response_code = wp_remote_retrieve_response_code( $wpcom_request );
+		if ( 200 !== $response_code ) {
+			return new WP_Error(
+				'failed_to_fetch_data',
+				esc_html__( 'Unable to fetch the requested data.', 'jetpack' ),
+				array( 'status' => $response_code )
+			);
+		}
+		return json_decode( wp_remote_retrieve_body( $wpcom_request ), true );
+	}
+
+	/**
+	 * Creates a Google Spreadsheet and returns some of its meta
+	 *
+	 * @param int    $user_id The user ID.
+	 * @param string $title   The spreadsheet title.
+	 * @param array  $rows    Array of arrays with values.
+	 * @return array
+	 */
+	public static function create_sheet( $user_id, $title, $rows = array() ) {
+		$site_id = self::get_site_id();
+		if ( is_wp_error( $site_id ) ) {
+			return false;
+		}
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			// check for gdrive helper class, call synchronously on .com
+			require_lib( 'google-sheets-helper' );
+			return wp_json_encode( WPCOM_Google_Sheets_Helper::get_create_spreadsheet_response( $user_id, $title, $rows ) );
+		}
+
+		$request_path  = sprintf( '/sites/%d/google-drive/sheets', $site_id );
+		$wpcom_request = Client::wpcom_json_api_request_as_user(
+			$request_path,
+			'2',
+			array(
+				'method'  => 'POST',
+				'headers' => array(
+					'content-type'    => 'application/json',
+					'X-Forwarded-For' => ( new Visitor() )->get_ip( true ),
+				),
+			),
+			array(
+				'parent_post_slug' => $title,
+				'rows'             => $rows,
+			)
+		);
+		$response_code = wp_remote_retrieve_response_code( $wpcom_request );
+		if ( 200 !== $response_code ) {
+			return new WP_Error(
+				'failed_to_fetch_data',
+				esc_html__( 'Unable to fetch the requested data.', 'jetpack' ),
+				array( 'status' => $response_code )
+			);
+		}
+		return json_decode( wp_remote_retrieve_body( $wpcom_request ), true );
```

Comment:
I'm not sure passing the response straight through is something we want to do even in this case. Reason being:

- Any code that calls this function now becomes closely coupled to whatever the WP.com API is returning. If you'd need to update either the API or the plugin, you potentially run into compatibility issues or unexpected side-effects.
- Another side effect is now there's not a single reference in Jetpack itself of what this function actually returns.

I'd say let's be explicit and 'manually' convert the response into an expected return value.